src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -383,11 +383,21 @@ public override Task<byte[]> GetArchiveHashAsync(HashAlgorithmName hashAlgorithm
 
         public override bool CanVerifySignedPackages(SignedPackageVerifierSettings verifierSettings)
         {
+            var value = false;
 #if IS_DESKTOP
-            return true;
-#else
-            return false;
+            if (RuntimeEnvironmentHelper.IsMono) {
+                if (RuntimeEnvironmentHelper.IsWindows) {
```

Comment:
nit: braces should be on a new line

Reply:
Removed and simplified with Damon's comment